crates/oxc_semantic/src/builder.rs

```diff
@@ -321,7 +321,17 @@ impl<'a> SemanticBuilder<'a> {
         excludes: SymbolFlags,
         report_error: bool,
     ) -> Option<SymbolId> {
-        let symbol_id = self.scope.get_binding(scope_id, name)?;
+        let symbol_id = self.scope.get_binding(scope_id, name).or_else(|| {
+            self.scope.get_parent_id(scope_id).and_then(|pid| {
+                let flags = self.scope.get_flags(pid);
+                // It is a Syntax Error if any element of the BoundNames of CatchParameter also occurs in the LexicallyDeclaredNames of Block.
```

Comment:
I don't think this check is in the right place 🤔

Reply:
I tried another way. Can you take a look at https://github.com/oxc-project/oxc/pull/4192/commits/3ed49909c2a226961b92601e11cee42108c4973a?